api/v1alpha1/nodepool_types.go

```diff
@@ -422,4 +422,8 @@ type AgentNodePoolPlatform struct {
 type AzureNodePoolPlatform struct {
 	VMSize  string `json:"vmsize"`
 	ImageID string `json:"imageID"`
+	// DiskSizeGB is the size in GB.
```

Comment:
thoughts about defaulting to 30GB here and make it explicit and predictable?

Reply:
Good idea, updated